File: backend/DTOs/CharacterDto.cs
using System.Text.Json.Serialization;

namespace PruebaTecnicaCarsales.DTOs
{
    public class CharacterDto
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public required string Status { get; set; }
        public required string Species { get; set; }
        public required string Type { get; set; }
        public required string Gender { get; set; }
        public required Origin Origin { get; set; }
        public required Location Location { get; set; }
        public required string Image { get; set; }
        [JsonPropertyName("episode")]
        public List<string> Episodes { get; set; } = [];
        public required string Url { get; set; }
        [JsonPropertyName("created")]
        public required string CreatedAt { get; set; }


    }

    public class Origin
    {
        public required string Name { get; set; }
        public required string Url { get; set; }
    }

    public class Location
    {
        public required string Name { get; set; }
        public required string Url { get; set; }
    }
}

File: frontend/src/app/features/episodes/models/dto/character.dto.ts
export interface CharacterDTO {
  name: string;
  status: string;
  species: string;
  type: string;
  gender: string;
  origin: OriginDTO;
  location: LocationDTO;
  image: string;
  episode: string[];
  url: string;
  created: string;
}

export interface LocationDTO {
  name: string;
  url: string;
}

export interface OriginDTO {
  name: string;
  url: string;
}
